_examples/checkout/main.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"gopkg.in/src-d/go-git.v4"
+	. "gopkg.in/src-d/go-git.v4/_examples"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+// Basic example of how to checkout a specific commit.
+func main() {
+	CheckArgs("<url>", "<directory>", "<commit-ref>")
+	url, directory, commitRef := os.Args[1], os.Args[2], os.Args[3]
+
+	// Clone the given repository to the given directory
+	Info("git clone %s %s --recursive", url, directory)
+
+	r, err := git.PlainClone(directory, false, &git.CloneOptions{
+		URL:               url,
+		RecurseSubmodules: git.DefaultSubmoduleRecursionDepth,
+	})
+
+	CheckIfError(err)
+
+	Info("git checkout %s", commitRef)
+
+	w, err := r.Worktree()
+
+	CheckIfError(err)
+
+	CheckIfError(w.Checkout(plumbing.NewHash(commitRef)))
+
+	fmt.Println("voila")
```

Comment:
@mcuadros, the problem as i stated in the pr description, is that the head commit will still be the old one, because go-git does not checkout the commit in a detached state. So you will actually still get the old HEAD commit, but the difference is that you will have the changes from the target-ref applied directly to your current branch . Therefore, printing the HEAD commit would be misleading. Let me know if there's a more proper way to do this. Thanks!